pkg/endpoint/public_ip.go

```diff
@@ -68,19 +98,23 @@ func getPublicIP(family k8snet.IPFamily, submSpec *types.SubmarinerSpecification
 	backendConfig map[string]string, airGapped bool,
 ) (string, string, error) {
 	switch family {
-	case k8snet.IPv4:
+	case k8snet.IPv4, k8snet.IPv6:
 		// If the node is annotated with a public-ip, the same is used as the public-ip of local endpoint.
 		config, ok := backendConfig[v1.PublicIP]
 		if !ok {
 			if submSpec.PublicIP != "" {
 				config = submSpec.PublicIP
 			} else {
-				config = getPublicIPResolvers()
+				config = getPublicIPResolvers(family)
 			}
 		}
+		// temporary workaround since GHA restricts IPv6 connections to external networks.
+		if family == k8snet.IPv6 {
+			airGapped = true
+		}
```

Comment:
Instead of forcing air gapped in code, can't we handle this in the GHA? I see there's an AIR_GAPPED defined in the shipyard Makefile.

Reply:
I added **air-gap** to (dual-stack, globalnet) and (dual-stack) gates ,  is that what you meant?
or there's another way to handle it in GHA ?

Also, I think we should also update Shipyard to block IPv6 traffic to external net by adding iptables rule (currently it is only implemented for IPv4) , but we can handle it in a follow up PR